src/riakc_pb_socket.erl

```diff
@@ -886,18 +886,30 @@ create_search_index(Pid, Index) ->
     create_search_index(Pid, Index, <<>>, []).
 
 %% @doc Create a search index.
--spec create_search_index(pid(), binary(), binary(), search_admin_opts()) ->
-                    ok | {error, term()}.
+
+-spec create_search_index(pid(), binary(), binary(),
+                          timeout()|search_admin_opts()) ->
+                                 ok | {error, term()}.
+create_search_index(Pid, Index, SchemaName, Timeout)
+  when is_integer(Timeout); Timeout =:= infinity  ->
+    create_search_index(Pid, Index, SchemaName, [{timeout, Timeout}]);
 create_search_index(Pid, Index, SchemaName, Opts) ->
     Timeout = proplists:get_value(timeout, Opts, default_timeout(search_timeout)),
     NVal = proplists:get_value(n_val, Opts),
-    Req = #rpbyokozunaindexputreq{
-        index = #rpbyokozunaindex{name = Index,
-                                  schema = SchemaName,
-                                  n_val = NVal}
-    },
-    call_infinity(Pid, {req, Req, Timeout}).
+    Req = set_search_req_nval(NVal, Index, SchemaName),
```

Comment:
This function is kind of badly named... you're not sending a search request, you're sending an "index creation" request.